core/domain/subtopic_page_services.py

```diff
@@ -207,6 +207,23 @@ def get_subtopic_pages_with_ids(
     return subtopic_pages
 
 
+def get_subtopic_pages_with_ids_and_versions(
+    topic_ids_subtopic_idxs_and_versions: List[Tuple[str, int, Optional[int]]]
+) -> List[Optional[subtopic_page_domain.SubtopicPage]]:
+    subtopic_page_ids_and_versions = [
+        (subtopic_page_domain.SubtopicPage.get_subtopic_page_id(
+            topic_id, subtopic_id), version)
```

Comment:
subtopic_index?

Reply:
The backend is a bit inconsistent with this, so I wasn't sure if I should continue that term outside of android.py (or if I should even use it at all). Sometimes it's subtopic_id, subtopic_index, or subtopic_page_id.

Reply:
Ah -- there are two different things here, not sure which you want.

subtopic_id is a unique ID for the subtopic. (Also called subtopic_page_id, but the domain object in topic_domain.py is Subtopic so subtopic_id is the more correct term.)
subtopic_index is the index of that subtopic in the list of subtopics. See get_subtopic_index() in topic_domain.py.

Not sure which of those you actually want. If subtopic_id then this probably makes sense to keep as subtopic_id but then I'm not sure if the arg to this function is correct.